diff --git a/package/mtk/applications/mtwifi-cfg-ucode/files/usr/share/ucode/mtwifi/converter.uc b/package/mtk/applications/mtwifi-cfg-ucode/files/usr/share/ucode/mtwifi/converter.uc
index 156b2667bb..8be2fce48d 100644
--- a/package/mtk/applications/mtwifi-cfg-ucode/files/usr/share/ucode/mtwifi/converter.uc
+++ b/package/mtk/applications/mtwifi-cfg-ucode/files/usr/share/ucode/mtwifi/converter.uc
@@ -134,20 +134,36 @@ export function convert(uci_cfg) {
 	let wmode_int = calc_wireless_mode(conf.band, conf.htmode);
 	dat.WirelessMode = wmode_int;
 	let is_ax = (wmode_int >= 16); // for TWT checking, etc.
+	let regions = (conf.country && length(conf.country) == 2) ? (defs.COUNTRY_REGIONS[conf.country] || [1, 0, "NA"]) : null;
 
 	// Channel, check auto or not
 	if (conf.channel == "auto") {
 		dat.AutoChannelSelect = "3";
 		dat.Channel = "0";
+
+		if (conf.band == "5g" && regions && regions[1] != null) {
+			let rd = regions[2];
+			if (rd && rd != "NA") {
+				dat.RDRegion = rd;
+				dat.IEEE80211H = "1";
+				dat.DfsEnable = "1";
+				dat.DfsDedicatedZeroWait = (bw_res.VHT_BW == "2") ? "3" : "2";
+				dat.DfsZeroWait = "1";
+				dat.DfsZeroWaitCacTime = "255";
+				dat.DfsZeroWaitDefault = "1";
+			} else {
+				dat.IEEE80211H = "0";
+				dat.DfsEnable = "0";
+			}
+		}
 	} else {
 		dat.AutoChannelSelect = "0";
 		dat.Channel = conf.channel;
 	}
 
 	// CountryRegion code
-	if (conf.country && length(conf.country) == 2) {
+	if (regions) {
 		dat.CountryCode = conf.country;
-		let regions = defs.COUNTRY_REGIONS[conf.country] || [1, 0];
 		if (conf.band == "2g") {
 			dat.CountryRegion = regions[0];
 		} else {
@@ -386,4 +402,4 @@ export function convert(uci_cfg) {
 	}
 
 	return dat;
-};
\ No newline at end of file
+};
