diff --git a/package/mtk/applications/mtwifi-cfg-ucode/files/lib/wifi/mtwifi.uc b/package/mtk/applications/mtwifi-cfg-ucode/files/lib/wifi/mtwifi.uc
index 5a377a0a0a..e71241c1d5 100755
--- a/package/mtk/applications/mtwifi-cfg-ucode/files/lib/wifi/mtwifi.uc
+++ b/package/mtk/applications/mtwifi-cfg-ucode/files/lib/wifi/mtwifi.uc
@@ -94,7 +94,7 @@ for (let devname in all_devnames) {
         "channel": "auto",
         "txpower": 100,
         "htmode": defs.htmode,
-        "country": "CN",
+        "country": "SE",
         "mu_beamformer": 1,
         "noscan": defs.htbsscoex,
         "serialize": 1
@@ -110,7 +110,8 @@ for (let devname in all_devnames) {
         "network": "lan",
         "mode": "ap",
         "ssid": defs.ssid,
-        "encryption": "none"
+        "encryption": "psk2",
+        "key": "password123"
     });
 
     need_commit = true;
